File: app/src/main/java/io/github/anonymous/pepper_realtime/ui/ChatMenuController.java
package io.github.anonymous.pepper_realtime.ui;

import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;

import com.google.android.material.navigation.NavigationView;

import io.github.anonymous.pepper_realtime.ui.ChatActivity;
import io.github.anonymous.pepper_realtime.R;
import io.github.anonymous.pepper_realtime.manager.DashboardManager;
import io.github.anonymous.pepper_realtime.manager.SettingsManager;

public class ChatMenuController {
    private final ChatActivity activity;
    private final DrawerLayout drawerLayout;
    private final MapUiManager mapUiManager;
    private final DashboardManager dashboardManager;
    private final SettingsManager settingsManager;

    public interface Listener {
        void onNewChatRequested();
    }

    private Listener listener;

    public ChatMenuController(ChatActivity activity, 
                            DrawerLayout drawerLayout, 
                            MapUiManager mapUiManager,
                            DashboardManager dashboardManager,
                            SettingsManager settingsManager) {
        this.activity = activity;
        this.drawerLayout = drawerLayout;
        this.mapUiManager = mapUiManager;
        this.dashboardManager = dashboardManager;
        this.settingsManager = settingsManager;
    }

    public void setListener(Listener listener) {
        this.listener = listener;
    }

    public void setupSettingsMenu() {
        NavigationView navigationView = activity.findViewById(R.id.navigation_view);
        // SettingsManager initialization delegated from Activity or passed in?
        // Assuming SettingsManager is already initialized in Activity and passed here
        
        drawerLayout.addDrawerListener(new DrawerLayout.SimpleDrawerListener() {
            @Override
            public void onDrawerClosed(@NonNull View drawerView) {
                if (settingsManager != null) {
                    settingsManager.onDrawerClosed();
                }
            }
        });
    }

    public boolean onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.chat_toolbar_menu, menu);
        return true;
    }

    public boolean onPrepareOptionsMenu(Menu menu) {
        MenuItem settingsItem = menu.findItem(R.id.action_settings);
        MenuItem newChatItem = menu.findItem(R.id.action_new_chat);
        MenuItem mapItem = menu.findItem(R.id.action_navigation_status);
        MenuItem dashboardItem = menu.findItem(R.id.action_dashboard);

        boolean isLandscape = activity.getResources().getConfiguration().orientation == 
            android.content.res.Configuration.ORIENTATION_LANDSCAPE;

        if (settingsItem != null) settingsItem.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
        if (newChatItem != null) newChatItem.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
        
        if (mapItem != null) {
            mapItem.setShowAsAction(isLandscape ? 
                MenuItem.SHOW_AS_ACTION_ALWAYS : MenuItem.SHOW_AS_ACTION_NEVER);
        }
        if (dashboardItem != null) {
            dashboardItem.setShowAsAction(isLandscape ? 
                MenuItem.SHOW_AS_ACTION_ALWAYS : MenuItem.SHOW_AS_ACTION_NEVER);
        }

        return true;
    }

    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == R.id.action_navigation_status) {
            mapUiManager.togglePreviewVisibility();
            mapUiManager.showNavigationStatusPopup();
            return true;
        } else if (itemId == R.id.action_new_chat) {
            if (listener != null) {
                listener.onNewChatRequested();
            }
            return true;
        } else if (itemId == R.id.action_dashboard) {
            if (dashboardManager != null) {
                dashboardManager.toggleDashboard();
            }
            return true;
        } else if (itemId == R.id.action_settings) {
            drawerLayout.openDrawer(GravityCompat.END);
            return true;
        }
        return false;
    }
}

File: app/src/main/java/io/github/anonymous/pepper_realtime/ui/ChatMenuController.kt
package io.github.anonymous.pepper_realtime.ui

import android.content.res.Configuration
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import io.github.anonymous.pepper_realtime.R
import io.github.anonymous.pepper_realtime.manager.DashboardManager
import io.github.anonymous.pepper_realtime.manager.SettingsManagerCompat

class ChatMenuController(
    private val activity: ChatActivity,
    private val drawerLayout: DrawerLayout,
    private val mapUiManager: MapUiManager,
    private val dashboardManager: DashboardManager?,
    private val settingsManager: SettingsManagerCompat?
) {

    fun interface Listener {
        fun onNewChatRequested()
    }

    private var listener: Listener? = null

    fun setListener(listener: Listener?) {
        this.listener = listener
    }

    fun setupSettingsMenu() {
        drawerLayout.addDrawerListener(object : DrawerLayout.SimpleDrawerListener() {
            override fun onDrawerOpened(drawerView: View) {
                settingsManager?.onDrawerOpened()
            }

            override fun onDrawerClosed(drawerView: View) {
                settingsManager?.onDrawerClosed()
            }
        })
    }

    fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater): Boolean {
        inflater.inflate(R.menu.chat_toolbar_menu, menu)
        return true
    }

    fun onPrepareOptionsMenu(menu: Menu): Boolean {
        val settingsItem = menu.findItem(R.id.action_settings)
        val newChatItem = menu.findItem(R.id.action_new_chat)
        val mapItem = menu.findItem(R.id.action_navigation_status)
        val dashboardItem = menu.findItem(R.id.action_dashboard)

        val isLandscape = activity.resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE

        settingsItem?.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS)
        newChatItem?.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS)

        mapItem?.setShowAsAction(
            if (isLandscape) MenuItem.SHOW_AS_ACTION_ALWAYS else MenuItem.SHOW_AS_ACTION_NEVER
        )
        dashboardItem?.setShowAsAction(
            if (isLandscape) MenuItem.SHOW_AS_ACTION_ALWAYS else MenuItem.SHOW_AS_ACTION_NEVER
        )

        return true
    }

    fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_navigation_status -> {
                mapUiManager.togglePreviewVisibility()
                mapUiManager.showNavigationStatusPopup()
                true
            }
            R.id.action_new_chat -> {
                listener?.onNewChatRequested()
                true
            }
            R.id.action_dashboard -> {
                dashboardManager?.toggleDashboard()
                true
            }
            R.id.action_settings -> {
                drawerLayout.openDrawer(GravityCompat.END)
                true
            }
            else -> false
        }
    }
}

